### ISSUE TITLE
[Feature Request] Manage 'Now Playing' queue

### ISSUE THREAD

**Issue by jolfrito** [digimezzo/dopamine-windows#521]
Hello!

I'd like to suggest a feature for those that always listen to their music library using the "Shuffle all" option. Actually this is one of the few things iTunes does right and it's worth imitating, so I'll explain what I mean using a screenshot from it:

![image](https://user-images.githubusercontent.com/32771771/31549521-1a487928-aff4-11e7-956f-9c457834a6cd.png)

As you can see, if you click the button circled in red, a small 'Up next' playlist appears. When you hover over a song, a red button (circled in yellow) appears so you can easily delete songs on the playlist. You can also rearrange songs by dragging and dropping.

I usually put some music on when showering using the "Shuffle all" option. I can't manually skip songs I don't want to listen to while I'm on the shower, so I just look at the 'Up next' playlist, delete the songs I don't want there and go on with my business. It's pretty straightforward.

Currently you can add songs to the "Now playing" queue in Dopamine, but you can't easily see what songs are going to play next, delete them or rearrange them. This is what it looks when you open "Now Playing":

![image](https://user-images.githubusercontent.com/32771771/31550416-e211496a-aff6-11e7-9da9-645773809e39.png)

As you can see, "Land" by Hollow Ran is playing, next to it comes "Honey Trap Aftermath" by Diablo Swing Orchestra (put there using the "play next" feature), but, after this song, the next songs on the "Shuffle All" queue don't appear, other songs by Hollow Ran come instead.

I'm not sure if i'm making myself clear, but I hope so.

tl;dr: it'd be cool to see what songs are coming next when using the "Shuffle All" feature, delete songs and rearrange the queue.

**Comment by Braillee7** [comment]
This is really missing in dopamine

**Comment by ZaffronSteam** [comment]
+1 To this, I feel like it might also help with an issue that's been brought up before <a href="https://github.com/digimezzo/dopamine-windows/issues/1069">#1069</a>.<br>Apologies for saddling a different issue but I do feel like having a "pre-shuffled" queue will remove the need to randomly jump between songs.

For context, I have a playlist with 180 songs and hitting shuffle causes it either to loop or repeat certain songs. I'm wondering if this is just the nature of true "randomness" rather than a bug and whether dopamine removes previously played songs from its shuffle queue. I feel like having a shuffle-all feature that creates a modifiable queue would force the player to follow a set list instead of randomly jumping from song to song with the potential to randomly repeat a song (apologies, this is just supposition btw).

Currently running Dopamine 2.0.9
Windows 11 V22621
